docs/upgrade.md

```diff
@@ -0,0 +1,20 @@
+# Upgrade
+
+This article walks you through how to upgrade your Coder server.
+
+## Upgrading Coder
+
```

Comment:
We oughta instruct people (loudly) that they should backup their database when upgrading a production deployment of Coder. We do not support rollbacks.